catroid/src/main/java/org/catrobat/catroid/content/Look.java

```diff
@@ -641,6 +645,13 @@ public void setHue(float hue) {
 		}
 	}
 
+	public void handleStopAllScripts() {
+		this.setWhenParallelAction(null);
+		Map<String, List<String>> scriptActions = new HashMap<>();
+		sprite.createStartScriptActionSequenceAndPutToMap(scriptActions, false);
+		StageActivity.stageListener.precomputeActionsForBroadcastEvents(scriptActions);
```

Comment:
I'd rather return the scriptActions from this method and call `precomputeActionsForBroadcastEvents()` from the `StopAllScriptsAction` since that already has the dependency to `StageActivity`. 
You would solve this without introducing the dependency here.

Just thinking: the Look shouldn't handle stage logic, the action does ;)